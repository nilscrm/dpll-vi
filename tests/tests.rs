use insta_cmd::assert_cmd_snapshot;
use rstest::rstest;
use std::{fs::File, path::{Path, PathBuf}, process::Command};

fn test_cnf_files(group_name: &str, path: &Path) {
  insta::with_settings!({
    snapshot_path => format!("snapshots/{}", group_name),
    filters => [(r"Performance\s+Time\s+.*\sms\s+Speed.*IPS", "[Performance...]")] },
  {
    let stdin = File::open(path).unwrap();
    assert_cmd_snapshot!(path.file_stem().unwrap().to_str().unwrap(), Command::new("vine").args(["run", "dpll/main.vi"]).arg("--breadth-first").stdin(stdin));
  });
}

#[rstest]
fn test_minimal(#[files("tests/minimal/*.cnf")] path: PathBuf) {
  test_cnf_files("minimal", &path);
}

#[rstest]
fn test_medium(#[files("tests/medium/*.cnf")] path: PathBuf) {
  test_cnf_files("medium", &path);
}

#[rstest]
fn test_satlib_sat(#[files("tests/uf5-218_first_10/*.cnf")] path: PathBuf) {
  test_cnf_files("uf5-218_first_10", &path);
}

#[rstest]
fn test_satlib_unsat(#[files("tests/uuf50-218_first_10/*.cnf")] path: PathBuf) {
  test_cnf_files("uuf50-218_first_10", &path);
}
