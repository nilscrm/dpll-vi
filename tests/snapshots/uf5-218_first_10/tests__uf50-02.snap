---
source: tests/tests.rs
info:
  program: vine
  args:
    - run
    - dpll/main.vi
    - "--breadth-first"
---
success: true
exit_code: 0
----- stdout -----
satisfiable
evaluation under model: true
var 1 = false
var 2 = true
var 3 = true
var 4 = false
var 5 = false
var 6 = true
var 7 = false
var 8 = false
var 9 = true
var 10 = true
var 11 = true
var 12 = false
var 13 = true
var 14 = true
var 15 = true
var 16 = true
var 17 = false
var 18 = true
var 19 = true
var 20 = false
var 21 = true
var 22 = true
var 23 = true
var 24 = false
var 25 = false
var 26 = true
var 27 = false
var 28 = false
var 29 = true
var 30 = false
var 31 = true
var 32 = false
var 33 = false
var 34 = true
var 35 = false
var 36 = true
var 37 = true
var 38 = true
var 39 = false
var 40 = false
var 41 = false
var 42 = true
var 43 = false
var 44 = true
var 45 = true
var 46 = false
var 47 = true
var 48 = true
var 49 = false
var 50 = false

----- stderr -----

Interactions
  Total           89_738_846
    Depth         45_671_675
    Breadth                1
  Annihilate      50_663_263
  Commute            143_895
  Copy             8_600_313
  Erase           10_270_442
  Expand           7_120_976
  Call             8_181_291
  Branch           4_758_666

Memory
  Heap               662_896 B
  Allocated    1_989_172_576 B
  Freed        1_989_172_576 B

[Performance...]
